File: android/src/main/java/aprakoso98/appsmgr/RNAppsManagerModule.java
package aprakoso98.appsmgr;

import android.content.pm.PackageManager;
import android.graphics.drawable.Drawable;
import android.util.Log;

import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.bridge.ReactContextBaseJavaModule;
import com.facebook.react.bridge.ReactMethod;
import com.facebook.react.bridge.Callback;
public class RNAppsManagerModule extends ReactContextBaseJavaModule {

    private final ReactApplicationContext reactContext;

    public RNAppsManagerModule(ReactApplicationContext reactContext) {
        super(reactContext);
        this.reactContext = reactContext;
    }

    @Override
    public String getName() {
        return "RNAppsManager";
    }

    @ReactMethod
    public void sampleMethod(String stringArgument, int numberArgument, Callback callback) {
        // TODO: Implement some actually useful functionality
        callback.invoke("Received numberArgument: " + numberArgument + " stringArgument: " + stringArgument);
    }
    @ReactMethod
    public void getIcon(String pkgName, Callback cb){
        try {
            Drawable icon = reactContext.getPackageManager().getApplicationIcon(pkgName);
            String iconB64 = Methods.Companion.encodeIcon(icon);
            cb.invoke("data:image/png;base64," + iconB64);
        } catch (PackageManager.NameNotFoundException e){
            Log.d("", pkgName);
        }
    }
}

File: android/src/main/java/aprakoso98/appsmgr/RNAppsManagerModule.kt
package aprakoso98.appsmgr

import com.facebook.react.bridge.*

class RNAppsManagerModule(private val context: ReactApplicationContext) : ReactContextBaseJavaModule(context) {

    override fun getName(): String {
        return "RNAppsManager"
    }

    @ReactMethod
    fun getIconFromPkgName(pkgName: String, promise: Promise) {
        try {
            val icon = Methods.getIconFromPkgName(pkgName, context)
            promise.resolve(icon)
        } catch (e: NotImplementedError){
            promise.reject(e)
        }
    }

    @ReactMethod
    fun getInformationApp(type: String = "", promise: Promise){
        try {
            val list = Methods.getAppInfo(context, type)
            promise.resolve(list)
        } catch (e: NotImplementedError){
            promise.reject(e)
        }
    }

    @ReactMethod
    fun killRunningProcess(pkgName: String, promise: Promise){
        try{
            Methods.killApp(context, pkgName)
            promise.resolve(true)
        } catch (e: NotImplementedError){
            promise.reject(e)
        }
    }
}
